app/lib/ui/metapolator/parameter-panel/selection/selection-parameter/selection-parameter-controller.js

```diff
@@ -1,16 +1,40 @@
 define([
-    'metapolator/ui/metapolator/ui-tools/selectionTools'
+      'metapolator/ui/metapolator/ui-tools/selectionTools'
+    , 'metapolator/errors'
+    , 'metapolator/math/simpleMathEngine'
+
 ], function(
-    selection
+      selection
+    , errors
+    , mathEngine
 ) {
     "use strict";
+
     function SelectionParameterController($scope, metapolatorModel, project) {
         $scope.selection = selection;
 
         $scope.changeValue = function(parameter, operator, value, event) {
-            var thisValue = manageInputValue(value, parameter, operator, event);
-            // Update the input
-            operator.low.current = thisValue;
+            var val = value.current;
+            if (typeof val === "string" && event.type === 'blur') {
+                val = evaluateMathExpression(val);
+                if (val[0]) {
+                    alert(val[0]);
+                    throw new errors.Input(val[0]);
```

Comment:
Is someone catching this or will it just bubble to the top? I suggest that at this point there should be a plan to handle bad input. I suggested something in this PR already.